packages/taiko-client/prover/proof_submitter/proof_submitter.go

```diff
@@ -134,6 +151,37 @@ func (s *ProofSubmitter) SubmitProof(
 		"tier", proofWithHeader.Tier,
 	)
 
+	// Check if we still need to generate a new proof for that block.
+	proofStatus, err := rpc.GetBlockProofStatus(ctx, s.rpc, proofWithHeader.BlockID, s.proverAddress)
+	if err != nil {
+		return err
+	}
+	if proofStatus.IsSubmitted {
+		return nil
+	}
+
+	if s.isGuardian {
+		_, expiredAt, _, err := handler.IsProvingWindowExpired(proofWithHeader.Meta, s.tiers)
+		if err != nil {
+			return fmt.Errorf("failed to check if the proving window is expired: %w", err)
+		}
+		// Get a random bumped submission delay, if necessary.
+		submissionDelay, err := s.getRandomBumpedSubmissionDelay(expiredAt)
+		if err != nil {
+			return err
+		}
+		// TODO:  Need to check if there will be performance issues while using simple time.Sleep()
+		time.Sleep(submissionDelay)
+		// Check again.
```

Comment:
Shall we only check here?

Reply:
If we only check here, sometimes we have to still wait until `expiredAt` + at least 1 hour when there's already a proof there. It's not efficient.

Reply:
I see